modules/EnsEMBL/Web/JSONServer/SpeciesSelector/Compara_Alignments.pm

```diff
@@ -57,18 +58,18 @@ sub json_fetch_species {
     $t->{children}   = [];
     my @children;
 
-    foreach (sort keys %{$row->{'species'}}) {
-      my $url_name  = $hub->species_defs->production_name_mapping($_);
-      my $prod_name = $hub->species_defs->get_config($url_name, 'SPECIES_PRODUCTION_NAME');
+    foreach $_ (sort keys %{$row->{'species'}}) {
+      my ($prod_name, $url_name); 
```

Comment:
is it better to keep these declarations before the loop starts?

Reply:
It's slightly better practice, yes - I guess i tend to do it because when you're assigning references to a variable you need to be more meticulous about scope. That's less critical here since we're only assigning strings.